addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -90,6 +103,153 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
         });
     },
 
+    /**
+     * Listeners for keyboard arrow / spacebar keys.
+     *
+     * - 39 = arrow-right
+     * - 32 = spacebar
+     * - 37 = arrow-left
+     *
+     * @param {KeyboardEvent} ev
+     */
+    _onKeyDown: function (ev) {
+        var keyCode = ev.keyCode;
+
+        if (keyCode === 39 || keyCode === 32) {
+            this._onNext(ev);
+        } else if (keyCode === 37) {
+            this._onBack(ev);
+        }
+    },
+
+    /**
+     * Handles the "next screen" behavior.
+     * It happens when the host uses the keyboard key / button to go to the next screen.
+     * The result depends on the current screen we're on.
+     *
+     * Possible values of the "next screen" to display are:
+     * - 'userInputs' when going from a question to the display of attendees' survey.user_input.line
+     *   for that question.
+     * - 'results' when going from the inputs to the actual correct / incorrect answers of that
+     *   question. Only used for scored simple / multiple choice questions.
+     * - 'leaderboard' (or 'leaderboardFinal') when going from the correct answers of a question to
+     *   the leaderboard of attendees. Only used for scored simple / multiple choice questions.
+     * - If it's not one of the above: we go to the next question, or end the session if we're on
+     *   the last question of this session.
+     *
+     * See '_getNextScreen' for a detailed logic.
+     *
+     * @param {Event} ev
+     */
+    _onNext: function (ev) {
+        ev.preventDefault();
+
+        this.currentScreen = this._getNextScreen();
+
+        if (this.currentScreen === 'userInputs') {
+            this._setShowInputs(true);
+            this.$('.o_survey_session_navigation_previous').removeClass('d-none');
+        } else if (this.currentScreen === 'results') {
+            this._setShowAnswers(true);
+            // when showing results, stop refreshing answers
+            clearInterval(this.resultsRefreshInterval);
+            delete this.resultsRefreshInterval;
+            this.$('.o_survey_session_navigation_previous').removeClass('d-none');
+        } else if (['leaderboard', 'leaderboardFinal'].includes(this.currentScreen)) {
+            if (this.currentScreen === 'leaderboardFinal') {
+                this.$('.o_survey_session_leaderboard_title').text(_('Final Leaderboard'));
+            }
+            this.leaderBoard.showLeaderboard();
+        } else {
+            if (this.isLastQuestion) {
+                this._endSession();
+            } else {
+                this._nextQuestion();
+            }
+        }
+    },
+
+    /**
+     * Reverse behavior of '_onNext'.
+     *
+     * @param {Event} ev
+     */
+    _onBack: function (ev) {
+        ev.preventDefault();
+
+        this.currentScreen = this._getPreviousScreen();
```

Comment:
Ok it is some kind of screen state actually currentScreen. We could imagine having var screenToDisplay, use it in computation, and finally setting this.currentScreen to screenToDisplay . That way it indicates we update the screen status of the widget. Not sure, just proposing. At first look I was a bit lost that currentScreen was set to getPreviouScreen (or NextScreen for the inverse method).